server/integration-tests/src/test/java/io/smallrye/graphql/tests/client/parsing/TypesafeClientMapTest.java

```diff
@@ -0,0 +1,215 @@
+        public ComplexToComplexMapWrapper complexToComplexWrapped(ComplexToComplexMapWrapper input) {
+            return input;
+        }
+
+    }
+
+    @GraphQLClientApi
+    public interface MapClientApi {
+
+        Map<Integer, String> scalarToScalar(Map<Integer, String> input);
+
+        Map<Foo, Integer> complexToScalar(Map<Foo, Integer> input);
+
+        Map<Integer, Foo> scalarToComplex(Map<Integer, Foo> input);
+
+        ComplexToComplexMapWrapper complexToComplexWrapped(ComplexToComplexMapWrapper input);
+
+    }
+
+    public static class ComplexToComplexMapWrapper {
+
+        private Map<Foo, Foo> map;
+
+        public Map<Foo, Foo> getMap() {
+            return map;
+        }
+
+        public void setMap(Map<Foo, Foo> map) {
+            this.map = map;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            ComplexToComplexMapWrapper that = (ComplexToComplexMapWrapper) o;
+            return Objects.equals(map, that.map);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(map);
+        }
+    }
+
+    public static class Foo {
+        private String bar;
+
+        public Foo() {
+        }
+
+        public Foo(String bar) {
+            this.bar = bar;
+        }
+
+        public String getBar() {
+            return bar;
+        }
+
+        public void setBar(String bar) {
+            this.bar = bar;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Foo foo = (Foo) o;
+            return Objects.equals(bar, foo.bar);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(bar);
+        }
+    }
+
+    @Test
+    public void scalarToScalar() {
+        Map<Integer, String> input = new HashMap<>();
+        input.put(1, "a");
+        input.put(2, "b");
+        Map<Integer, String> result = client.scalarToScalar(input);
+        assertEquals("a", result.get(1));
+        assertEquals("b", result.get(2));
+        assertEquals(2, result.size());
+    }
+
+    @Test
+    public void complexToScalar() {
+        try {
+            Map<Foo, Integer> input = new HashMap<>();
+            input.put(new Foo("a"), 68);
+            input.put(new Foo("x"), 55);
+            Map<Foo, Integer> result = client.complexToScalar(input);
+            assertEquals(68L, result.get(new Foo("a")).longValue());
+            assertEquals(55L, result.get(new Foo("x")).longValue());
+            assertEquals(2, result.size());
+        } catch (Throwable t) {
+            t.printStackTrace();
+            throw t;
+        }
```

Comment:
What kinds of exceptions are being ignored here?

Reply:
Nothing is being ignored, I think added that because without `t.printStackTrace()`, what appeared in the log did not contain all the relevant info (in the case it is a `GraphQLClientException`, the actual errors)

Reply:
Sorry, I just did read the `t.printStackTrace()`.
But exceptions thrown in a test should be logged incl. their stack trace! Aren't they?

Reply:
I don't remember exactly but I think the stack trace was there, but the message wasn't, I couldn't tell the graphql level errors from the log